addons/medical_gui/functions/fnc_onKeyDown.sqf

```diff
@@ -0,0 +1,84 @@
+#include "..\script_component.hpp"
+#include "\a3\ui_f\hpp\defineDIKCodes.inc"
+/*
+ * Author: AmsteadRayle
+ * Handles keyboard inputs in medical menu.
+ *
+ * Arguments:
+ * 0: Menu display <DISPLAY>
+ * 1: Key being pressed <NUMBER>
+ * 2: Shift state <BOOL>
+ * 3: Ctrl state <BOOL>
+ * 4: Alt state <BOOL>
+ *
+ * Return Value:
+ * None
+ *
+ * Public: No
+*/
+// TODO: Is the airway category ever visible? Can the dynamic category stuff be removed?
+
+#define NUMBER_KEYS [DIK_1, DIK_2, DIK_3, DIK_4, DIK_5, DIK_6, DIK_7, DIK_8, DIK_9, DIK_0]
+
+params ["", "_args"];
+_args params ["_display", "_keyPressed", "_shiftState", "_ctrlState", "_altState"];
+
+private _return = true; // Override existing keybinds for keys used here
+
+private _visibleCategories = [
+    "bandage","medication","airway","advanced","drag"
+] select {
+    private _category = _x;
+    (GVAR(actions) findIf {_category == _x select 1}) > -1
+};
+
+private _allCategories = ["triage", "examine"] + _visibleCategories + ["toggle"];
+
+// Use hashmap as a shortcut to "zip" two arrays together
+// Use categories as keys in hashmap because there are fewer,
+// otherwise the hashmap is padded with nil
+private _keyCategoryPairs = _allCategories createHashMapFromArray NUMBER_KEYS;
+
+private _temp_category = "";
+
+switch (true) do {
```

Comment:
Can't this just be `switch _keyPressed`?

Reply:
Probably. But it looks like `switch (true)` allows for using modifier keys, looking at the arsenal implementation, which this is drawn from.
https://github.com/acemod/ACE3/blob/176482381ad2222502157933c0cd622f2a049da2/addons/arsenal/functions/fnc_onKeyDown.sqf#L32-L41

Reply:
Are modifier keys going to be used here though? `_keyPressed` is still limited to DIK key codes.